shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -554,23 +559,20 @@ impl CassandraCodec2 {
         }
     }
 
-    fn decode_raw(&mut self, src: &mut BytesMut) -> Result<Option<Frame>> {
-        trace!("Parsing C* frame");
-        let v = parser::parse_frame(src, &self.compressor, &self.current_head);
-        match v {
-            Ok((r, h)) => {
-                self.current_head = h;
-                Ok(r)
-            }
-            // Note these should be parse errors, not actual protocol errors
-            Err(e) => Err(anyhow!(e)),
-        }
+    fn decode_raw(&mut self, src: &mut BytesMut) -> Result<Option<Frame>, CDRSError> {
+        trace!("{:?} Parsing C* frame", thread::current().id());
+        let v: Result<(Option<Frame>, Option<FrameHeader>), CDRSError> =
+            parser::parse_frame(src, &self.compressor, self.current_head.as_ref());
+        v.map(|(r, h)| {
+            self.current_head = h;
+            r
+        })
     }
 
     fn encode_raw(&mut self, item: Frame, dst: &mut BytesMut) {
         let buffer = item.into_cbytes();
         if buffer.is_empty() {
-            info!("trying to send 0 length frame");
+            info!("{:?} trying to send 0 length frame", thread::current().id());
```

Comment:
Im not really sure what all the `thread::current().id()` changes are for.
Maybe move it into a separate PR and make a case for why we need to know the opaque thread id of the thread that each log came from.

Reply:
There is a proper fix coming in https://github.com/shotover/shotover-proxy/pull/363 so lets remove those changes.

Reply:
Looks fine to me
![image](https://user-images.githubusercontent.com/5120858/141376904-d03ac1d9-d914-435c-9d18-69c22617ff9c.png)

The reason we need to remove it is because:
* Its adding a lot of noise to the PR and making it harder to review the real changes
* Its adding misleading information to the logs, thread id is an incorrect abstraction in an async world.

Reply:
yup lets pull it out of this change - knowing the thread for a log statement is good, but should be in a seperate PR

Reply:
> In your example, why does the WARN statement (2nd line) not contain the Request info?

Because your PR removes the line that passes the span into that task. (I added a comment to direct you to the exact line)

> Are we assuming that two threads will not execute for the same Request at the same time?

In general yes, the message has to finish processing in one task before it can be sent onto the next one.

This assumption is not true for something like Tee or ParallelMap where the message is cloned and sent down multiple chains. For those more work needs to be done to spans so we can distinguish between the branching chains.

However your PR does not touch those transforms, so for the purposes of your PR you can assume "yes".